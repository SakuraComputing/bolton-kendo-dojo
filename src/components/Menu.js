import React from 'react'
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { logoutUser } from '../actions/authActions';

export class Menu extends React.Component {

  onLogoutClick = (e) => {
      e.preventDefault();
      this.props.logoutUser();
  }  

  render() { 

        const { isAuthenticated } = this.props.auth;

        const authLinks = (
            <div>
                <li>
                    <Link to="/study">Kendo Study</Link>
                </li>
                <li>
                    <Link
                        to="/members"
                    >
                    Members
                    </Link>
                </li>
                <li>
                    <Link 
                        to="/logout"
                        href=""
                        onClick={this.onLogoutClick}
                    >
                    Logout
                    </Link>
                </li>
            </div>
        )

        const guestLinks = (
            <li>
                <Link to="/login">Login / Registration</Link>
            </li>
        )

        return (
            <div>
                <ul className="nav">
                    <li>
                        <Link to="/">Home</Link>
                    </li>
                    <li>
                        <Link to="/about">Where and When</Link>
                    </li>
                    <li>
                        <Link to="/history">Club History</Link>
                    </li>
                    <li>
                        <Link to="/theory">What is Kendo</Link>
                    </li>
                    <li>
                        <Link to="/links">Kendo Links</Link>
                    </li>
                    <li>
                        <Link to="/photos">Kendo Photograph's</Link>
                    </li>
                    <li>
                        <Link to="/videos">Kendo Videos</Link>
                    </li>
                    {isAuthenticated ? authLinks :  guestLinks}
                </ul>
            </div>
        )
    }
}
const mapStateToProps = state => ({
    auth: state.auth
})
export default connect(mapStateToProps, { logoutUser})(Menu);
