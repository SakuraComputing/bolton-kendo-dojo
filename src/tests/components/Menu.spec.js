import { shallow } from 'enzyme';
import React from 'react';
import { Menu } from '../../components/Menu.js'

describe('menu component', () => {

    let wrapper, auth;

    beforeEach(() => {
        auth = jest.fn();
        wrapper = shallow(<Menu auth={auth} />);
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should render different options when user logged in', () => {
        expect(wrapper).toMatchSnapshot();        
    });
});