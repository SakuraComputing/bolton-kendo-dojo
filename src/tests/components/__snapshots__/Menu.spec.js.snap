// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    auth={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="nav"
      >
        <li>
          <Link
            replace={false}
            to="/"
          >
            Home
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/about"
          >
            Where and When
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/calendar"
          >
            Calendar
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/history"
          >
            Club History
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/theory"
          >
            What is Kendo
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/links"
          >
            Kendo Links
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/media"
          >
            Kendo Media
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/login"
          >
            Login / Registration
          </Link>
        </li>
      </ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <Link
              replace={false}
              to="/"
            >
              Home
            </Link>
          </li>,
          <li>
            <Link
              replace={false}
              to="/about"
            >
              Where and When
            </Link>
          </li>,
          <li>
            <Link
              replace={false}
              to="/calendar"
            >
              Calendar
            </Link>
          </li>,
          <li>
            <Link
              replace={false}
              to="/history"
            >
              Club History
            </Link>
          </li>,
          <li>
            <Link
              replace={false}
              to="/theory"
            >
              What is Kendo
            </Link>
          </li>,
          <li>
            <Link
              replace={false}
              to="/links"
            >
              Kendo Links
            </Link>
          </li>,
          <li>
            <Link
              replace={false}
              to="/media"
            >
              Kendo Media
            </Link>
          </li>,
          <li>
            <Link
              replace={false}
              to="/login"
            >
              Login / Registration
            </Link>
          </li>,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
            >
              Home
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Home",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/about"
            >
              Where and When
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Where and When",
              "replace": false,
              "to": "/about",
            },
            "ref": null,
            "rendered": "Where and When",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/calendar"
            >
              Calendar
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Calendar",
              "replace": false,
              "to": "/calendar",
            },
            "ref": null,
            "rendered": "Calendar",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/history"
            >
              Club History
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Club History",
              "replace": false,
              "to": "/history",
            },
            "ref": null,
            "rendered": "Club History",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/theory"
            >
              What is Kendo
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "What is Kendo",
              "replace": false,
              "to": "/theory",
            },
            "ref": null,
            "rendered": "What is Kendo",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/links"
            >
              Kendo Links
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Kendo Links",
              "replace": false,
              "to": "/links",
            },
            "ref": null,
            "rendered": "Kendo Links",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/media"
            >
              Kendo Media
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Kendo Media",
              "replace": false,
              "to": "/media",
            },
            "ref": null,
            "rendered": "Kendo Media",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/login"
            >
              Login / Registration
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Login / Registration",
              "replace": false,
              "to": "/login",
            },
            "ref": null,
            "rendered": "Login / Registration",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="nav"
        >
          <li>
            <Link
              replace={false}
              to="/"
            >
              Home
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/about"
            >
              Where and When
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/calendar"
            >
              Calendar
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/history"
            >
              Club History
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/theory"
            >
              What is Kendo
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/links"
            >
              Kendo Links
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/media"
            >
              Kendo Media
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/login"
            >
              Login / Registration
            </Link>
          </li>
        </ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Link
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/about"
              >
                Where and When
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/calendar"
              >
                Calendar
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/history"
              >
                Club History
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/theory"
              >
                What is Kendo
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/links"
              >
                Kendo Links
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/media"
              >
                Kendo Media
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/login"
              >
                Login / Registration
              </Link>
            </li>,
          ],
          "className": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
              >
                Home
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Home",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/about"
              >
                Where and When
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Where and When",
                "replace": false,
                "to": "/about",
              },
              "ref": null,
              "rendered": "Where and When",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/calendar"
              >
                Calendar
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Calendar",
                "replace": false,
                "to": "/calendar",
              },
              "ref": null,
              "rendered": "Calendar",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/history"
              >
                Club History
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Club History",
                "replace": false,
                "to": "/history",
              },
              "ref": null,
              "rendered": "Club History",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/theory"
              >
                What is Kendo
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "What is Kendo",
                "replace": false,
                "to": "/theory",
              },
              "ref": null,
              "rendered": "What is Kendo",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/links"
              >
                Kendo Links
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Kendo Links",
                "replace": false,
                "to": "/links",
              },
              "ref": null,
              "rendered": "Kendo Links",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/media"
              >
                Kendo Media
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Kendo Media",
                "replace": false,
                "to": "/media",
              },
              "ref": null,
              "rendered": "Kendo Media",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/login"
              >
                Login / Registration
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Login / Registration",
                "replace": false,
                "to": "/login",
              },
              "ref": null,
              "rendered": "Login / Registration",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`menu component should render different options when user logged in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    auth={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="nav"
      >
        <li>
          <Link
            replace={false}
            to="/"
          >
            Home
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/about"
          >
            Where and When
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/calendar"
          >
            Calendar
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/history"
          >
            Club History
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/theory"
          >
            What is Kendo
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/links"
          >
            Kendo Links
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/media"
          >
            Kendo Media
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/login"
          >
            Login / Registration
          </Link>
        </li>
      </ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <Link
              replace={false}
              to="/"
            >
              Home
            </Link>
          </li>,
          <li>
            <Link
              replace={false}
              to="/about"
            >
              Where and When
            </Link>
          </li>,
          <li>
            <Link
              replace={false}
              to="/calendar"
            >
              Calendar
            </Link>
          </li>,
          <li>
            <Link
              replace={false}
              to="/history"
            >
              Club History
            </Link>
          </li>,
          <li>
            <Link
              replace={false}
              to="/theory"
            >
              What is Kendo
            </Link>
          </li>,
          <li>
            <Link
              replace={false}
              to="/links"
            >
              Kendo Links
            </Link>
          </li>,
          <li>
            <Link
              replace={false}
              to="/media"
            >
              Kendo Media
            </Link>
          </li>,
          <li>
            <Link
              replace={false}
              to="/login"
            >
              Login / Registration
            </Link>
          </li>,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
            >
              Home
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Home",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/about"
            >
              Where and When
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Where and When",
              "replace": false,
              "to": "/about",
            },
            "ref": null,
            "rendered": "Where and When",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/calendar"
            >
              Calendar
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Calendar",
              "replace": false,
              "to": "/calendar",
            },
            "ref": null,
            "rendered": "Calendar",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/history"
            >
              Club History
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Club History",
              "replace": false,
              "to": "/history",
            },
            "ref": null,
            "rendered": "Club History",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/theory"
            >
              What is Kendo
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "What is Kendo",
              "replace": false,
              "to": "/theory",
            },
            "ref": null,
            "rendered": "What is Kendo",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/links"
            >
              Kendo Links
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Kendo Links",
              "replace": false,
              "to": "/links",
            },
            "ref": null,
            "rendered": "Kendo Links",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/media"
            >
              Kendo Media
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Kendo Media",
              "replace": false,
              "to": "/media",
            },
            "ref": null,
            "rendered": "Kendo Media",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/login"
            >
              Login / Registration
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Login / Registration",
              "replace": false,
              "to": "/login",
            },
            "ref": null,
            "rendered": "Login / Registration",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="nav"
        >
          <li>
            <Link
              replace={false}
              to="/"
            >
              Home
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/about"
            >
              Where and When
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/calendar"
            >
              Calendar
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/history"
            >
              Club History
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/theory"
            >
              What is Kendo
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/links"
            >
              Kendo Links
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/media"
            >
              Kendo Media
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/login"
            >
              Login / Registration
            </Link>
          </li>
        </ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Link
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/about"
              >
                Where and When
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/calendar"
              >
                Calendar
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/history"
              >
                Club History
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/theory"
              >
                What is Kendo
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/links"
              >
                Kendo Links
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/media"
              >
                Kendo Media
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/login"
              >
                Login / Registration
              </Link>
            </li>,
          ],
          "className": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
              >
                Home
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Home",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/about"
              >
                Where and When
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Where and When",
                "replace": false,
                "to": "/about",
              },
              "ref": null,
              "rendered": "Where and When",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/calendar"
              >
                Calendar
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Calendar",
                "replace": false,
                "to": "/calendar",
              },
              "ref": null,
              "rendered": "Calendar",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/history"
              >
                Club History
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Club History",
                "replace": false,
                "to": "/history",
              },
              "ref": null,
              "rendered": "Club History",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/theory"
              >
                What is Kendo
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "What is Kendo",
                "replace": false,
                "to": "/theory",
              },
              "ref": null,
              "rendered": "What is Kendo",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/links"
              >
                Kendo Links
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Kendo Links",
                "replace": false,
                "to": "/links",
              },
              "ref": null,
              "rendered": "Kendo Links",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/media"
              >
                Kendo Media
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Kendo Media",
                "replace": false,
                "to": "/media",
              },
              "ref": null,
              "rendered": "Kendo Media",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/login"
              >
                Login / Registration
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Login / Registration",
                "replace": false,
                "to": "/login",
              },
              "ref": null,
              "rendered": "Login / Registration",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
